README.md

```diff
@@ -174,7 +176,7 @@ These are the concrete steps to follow to move your code from branch to branch:
    `develop` branch.
    - Consider doing a
      [squash-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits),
-     especially if your pull request is relatively small, to keep a
+     except if your pull request is relatively small, to keep a
```

Comment:
Good point -- so, I'd think we'd want to squash all merges to `develop`, then.

My reasoning for squashing aggressively is because we're working on GitHub, where the original chain of commits is retained in the closed PR, even when what gets merged to the base branch is a single squash.

It has the benefit of clean base branch history, while still retaining the full arc of the work process in the PR.

So, I'm inclined to strengthen this to something like "Please use GitHub's squash-merge functionality for all PRs to `develop`."

---


For merges to `main`, since release/hotfix branches can have weird history topology (e.g., a release gets hotfixed, then merges to `main`, then merges to `develop`) that would be hard to follow if it got squash-merged to either `main` or `develop`, I'm inclined to do simple commit merges there.

Reply:
I'll write out my case in a new commit here, see what you think on your next review pass.

Reply:
+1 to all of that